include/eve/detail/function/simd/common/swap_adjacent_group.hpp

```diff
@@ -0,0 +1,49 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/pattern.hpp>
+
+namespace eve::detail
+{
+  template<simd_value Wide, std::ptrdiff_t G>
+  EVE_FORCEINLINE auto swap_adjacent_group_(EVE_SUPPORTS(cpu_), Wide v, fixed<G>) noexcept
+  requires(G <= Wide::size() )
+  {
+    if constexpr(G == Wide::size())
+    {
+      return v;
+    }
+    else if constexpr( has_aggregated_abi_v<Wide> )
+    {
+      if constexpr(G == Wide::size()/2)
+      {
+        auto[l,h] = v.slice();
+        return Wide{h,l};
+      }
+      else
+      {
+        auto[l,h] = v.slice();
+        return Wide {swap_adjacent_group(l,fixed<G>{}),swap_adjacent_group(h,fixed<G>{})};
+      }
+    }
+    else
+    {
+      // In this case, we don't have anything special to do so we just swizzle
+      return v[eve::swap_adjacent_group_n<G,Wide::size()>];
+    }
```

Comment:
Hmm... 
Falling back to this swizzle means:
1) we are going through compilation again. Which means we are going to cycle (since you will add this shuffle to it).
2) we don't want to compile all of that anyways.

Also you don't want to do that. You want to double size while you can. And then do the very basic swizzle and not this swizzle.

These individual swizzles should know their optimal implementations and not recompile.

Reply:
OK

Reply:
In fact, nobody should fell here anyway except for emulation.
So I guess I just call basic_swizzle directly instead of []